yasnippet.el

```diff
@@ -3467,17 +3467,23 @@ Move the overlays, or create them if they do not exit."
              ;; (overlay-put ov 'evaporate t)
              (overlay-put ov 'modification-hooks '(yas--on-protection-overlay-modification)))))))
 
-(defun yas--on-protection-overlay-modification (_overlay after? _beg _end &optional _length)
+(defvar yas--buffer-pre-modification)
```

Comment:
In those cases it appears that the modification hook is called twice:

```
(defun joaot/check (ov afterp beg end &optional length))

(with-temp-buffer
  (insert "aaabbbccc")
  (let ((ov (make-overlay 3 6)))
    (overlay-put ov 'modification-hooks
                 '(joaot/check)))
  (delete-region 4 5)
  (goto-char 4)
  (insert "z"))
```

Now trace `joaot/check` and look at `*trace-output*`:

```
======================================================================
1 -> (joaot/check #<overlay from 3 to 6 in  *temp*> nil 4 5)
1 <- joaot/check: nil
======================================================================
1 -> (joaot/check #<overlay from 3 to 5 in  *temp*> t 4 4 1)
1 <- joaot/check: nil
======================================================================
1 -> (joaot/check #<overlay from 3 to 5 in  *temp*> nil 4 4)
1 <- joaot/check: nil
======================================================================
1 -> (joaot/check #<overlay from 3 to 6 in  *temp*> t 4 5 0)
1 <- joaot/check: nil
```

The second entry would raise the alarm bells and commit the snippet.